master/internal/db/setup.go

```diff
@@ -16,6 +19,51 @@ const (
 	sslTpl = "&sslmode=%s&sslrootcert=%s"
 )
 
+// authTokenKeypair gets the existing auth token keypair.
+func authTokenKeypair(ctx context.Context) (*model.AuthTokenKeypair, error) {
+	var tokenKeypair model.AuthTokenKeypair
+	switch err := Bun().NewSelect().Table("auth_token_keypair").Scan(ctx, &tokenKeypair); {
+	case errors.Cause(err) == ErrNotFound:
+		return nil, nil
+	case err != nil:
+		return nil, err
+	default:
+		return &tokenKeypair, nil
+	}
+}
+
+// addAuthTokenKeypair adds the new auth token keypair.
+func addAuthTokenKeypair(ctx context.Context, tokenKeypair *model.AuthTokenKeypair) error {
+	newTok := &model.AuthTokenKeypair{
+		PublicKey:  tokenKeypair.PublicKey,
+		PrivateKey: tokenKeypair.PrivateKey,
+	}
+	_, err := Bun().NewInsert().Model(newTok).Exec(ctx)
+	return err
+}
+
+// InitAuthKeys initializes auth token keypairs.
+func InitAuthKeys() error {
+	switch storedKeys, err := authTokenKeypair(context.TODO()); {
+	case err != nil:
+		return errors.Wrap(err, "error retrieving auth token keypair")
+	case storedKeys == nil:
+		publicKey, privateKey, err := ed25519.GenerateKey(nil)
+		if err != nil {
+			return errors.Wrap(err, "error creating auth token keypair")
+		}
+		tokenKeypair := model.AuthTokenKeypair{PublicKey: publicKey, PrivateKey: privateKey}
+		err = addAuthTokenKeypair(context.TODO(), &tokenKeypair)
+		if err != nil {
+			return errors.Wrap(err, "error saving auth token keypair")
+		}
+		SetTokenKeys(&tokenKeypair)
+	default:
+		SetTokenKeys(storedKeys)
+	}
+	return nil
+}
+
```

Comment:
Moved `authTokenKeypair`, `addAuthTokenKeypair`, and `InitAuthKeys` to this file because, if moved to `user/postgres_users.go` they would create an import cycle between the db and user packages. Happy to take suggestions on where else to move this.